### PR TITLE
[flutter_tools] add debugging to ios/core_devices.dart

### PR DESCRIPTION
**Pull request by christopherfujino**
Add debugging for #141892 to detect when the temp file mysteriously disappears after running devicectl.

### CODE REVIEW

**Comment by christopherfujino** [comment]
FYI @eliasyishak, please let me know if you see this test fail after this lands

**Comment by vashworth** [comment]
Looks like you hit the issue in presubmit: https://ci.chromium.org/ui/p/flutter/builders/try/Mac%20tool_integration_tests_4_4/32144/overview

It doesn't print out the print trace, only the StateError, probably because the command wasn't verbose?

File: packages/flutter_tools/test/integration.shard/overall_experience_test.dart [hunk 1]
```diff
@@ -108,16 +117,22 @@ void main() {
...
+          'Performing hot restart...'.padRight(progressMessageWidth),
+          startsWith('Restarted application in '),
+          '', // this newline is the one for after we hit "q"
+          'Application finished.',
+          'ready',
+        ],
+        reason: 'stdout from command ${command.join(' ')} was unexpected, '
+          'full Stdout:\n\n${result.stdout.join('\n')}\n\n'
+          'Stderr:\n\n${result.stderr.join('\n')}',
+      );
```
Comment (eliasyishak) [inline packages/flutter_tools/test/integration.shard/overall_experience_test.dart:135]: I've also been seeing the test below this one failing with the same error, might want to add it there too?

`flutter run can hot reload and hot restart, handle "p" key`

**Comment by christopherfujino** [comment]
> It doesn't print out the print trace, only the StateError, probably because the command wasn't verbose?

ahh, you're right, and the whole point of this test is to verify the non-verbose output.

**Comment by christopherfujino** [comment]
Ok, pushed a change commenting out every test except the one in question, and running it 100 times. hopefully we'll get a repro with the full stacktrace.

File: packages/flutter_tools/test/integration.shard/overall_experience_test.dart (hunk 1 above)
Comment (christopherfujino) [reply packages/flutter_tools/test/integration.shard/overall_experience_test.dart]: ack, and done

**Comment by christopherfujino** [comment]
> Ok, pushed a change commenting out every test except the one in question, and running it 100 times. hopefully we'll get a repro with the full stacktrace.

for whatever reason, running just the test in question 100x across several builds did not yield the error (it might have to do with running all the tests sequentially), so I moved the interesting prints from trace to error, which the test expectation should print when it fails.

**Review by andrewkolos — APPROVED** [review]
RSLGTM

**Pull request merged**